opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -2343,8 +2384,44 @@ namespace detail {
     }
 
 
+    template <class Grid, class Implementation>
+    double
+    BlackoilModelBase<Grid, Implementation>::
+    computeTimeError(const SimulatorState& previous,
+                     const SimulatorState& current ) const
+    {
+        std::vector< double > p0  ( previous.pressure() );
+        std::vector< double > sat0( previous.saturation() );
+
+        const std::size_t pSize = p0.size();
+        const std::size_t satSize = sat0.size();
+
+        // compute u^n - u^n+1
+        for( std::size_t i=0; i<pSize; ++i ) {
+            p0[ i ]   -= current.pressure()[ i ];
+        }
 
+        for( std::size_t i=0; i<satSize; ++i ) {
+            sat0[ i ] -= current.saturation()[ i ];
+        }
+
+        // compute || u^n - u^n+1 ||
+        const double stateOld  = detail::euclidianNormSquared( p0.begin(),   p0.end(), 1, linsolver_.parallelInformation() ) +
+                                 detail::euclidianNormSquared( sat0.begin(), sat0.end(),
+                                                               current.numPhases(),
+                                                               linsolver_.parallelInformation() );
 
+        // compute || u^n+1 ||
```

Comment:
There is a complete lack of scaling here, so this will be dominated by the pressure. I assume that is how the current code works if I interpreted you correctly?

If so I'm fine with changing this later as you suggest.